pandas/core/frame.py

```diff
@@ -7022,56 +7022,154 @@ def cov(self, min_periods=None):
 
         return self._constructor(baseCov, index=idx, columns=cols)
 
-    def corrwith(self, other, axis=0, drop=False):
+    def corrwith(self, other, axis=0, how='pairwise', drop=False,
+                 method='pearson', min_periods=1):
         """
-        Compute pairwise correlation between rows or columns of two DataFrame
-        objects.
+        Compute correlation between rows or columns of two DataFrame objects
+        or DataFrame and Series objects.
 
         Parameters
         ----------
         other : DataFrame, Series
         axis : {0 or 'index', 1 or 'columns'}, default 0
             0 or 'index' to compute column-wise, 1 or 'columns' for row-wise
+        how : {'pairwise', 'all'}, default 'pairwise' if other is a DataFrame
+            and default 'all' if other is a Series object
+            * 'pairwise' to compute correlation between rows or columns with
```

Comment:
this is a really confusing default

Reply:
Want to break the backward compatibility?

Reply:
you haven't answered the question